contents/javascript/Javascript-promises.md

```diff
@@ -89,58 +87,117 @@ These situations correspond to the 3 states of Promises in Javascript.
 
 Promises provide the ability to specify how the execution of some part of your code would depend on the status of an asynchronous operation. It can now wait for the asynchronous operation to resolve first before doing any work on its result.
 
-## Quickstart
+## Getting Started 
+
+Here, we will be showing you some sample code to get started with promises. Let's learn how to go from callbacks to promises using a few examples.
+
+### From callbacks to promises
+Say we have two time intensive functions, `getData()` and `filterData()` which require some time to complete. You will have to get the data from some server using `getData()`, then process it using `filterData()`, before you can start displaying the results.
+
+How would such functions be implemented using the callback method? The callback method utilizes the fact that we can easily pass functions into javascript as parameters and then use them within the function, effectively "passing" any form of data out, without explicitly returning any value.
 
-Let's return to the setTimeout example at the beginning and rewrite that with promises. 
+We will implement the above use case in the "callback style" (mimicking the long return time of the functions using `setTimeout`):
 
 ```javascript
+function executeWithDelay(val, callback) {
+    // return val after a short wait
+    setTimeout(function(){
+        callback(val);
+    }, 1000);
+}
+
+/** The very long calls **/
+function getData(callback) {
+    executeWithDelay('some random data', callback);
+}
+
+function filterData(data, callback) {
+    executeWithDelay(data.split(' '), callback);
+}
+
+function main() {
+    getData(function(data){
+        filterData(function(filtered){
+            // will print array of splitted text
+            console.log(filtered);
+        })
+    });
+}
+
+main();
+```
+
+Now we will rewrite all these using promises. We will be using the same function and variable names, this will let you see how exactly promise works, as opposed to the callback style that you might be more familar with.
```

Comment:
> function and variable names, this will let you see how exactly promise works, as opposed to the callback style that you might be more familar with

->

function and variable names, to show how exactly promises compare to callbacks.